src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java

```diff
@@ -300,7 +302,10 @@ private static ModulePropertyDetails createProperties(DetailNode nodeLi) {
             final DetailNode propertyNameTag = propertyNameNode.get();
             final String propertyName = getTextFromTag(propertyNameTag);
 
-            final DetailNode propertyType = getFirstChildOfMatchingText(nodeLi, TYPE_TAG)
+            @Present final Optional<DetailNode> firstChildOfMatchingText =
+                getFirstChildOfMatchingText(nodeLi, TYPE_TAG);
+
+            final DetailNode propertyType = firstChildOfMatchingText
                 .orElseThrow(() -> {
```

Comment:
I am beginning to understand this Checker a bit more... our code does not make sense. If we are guaranteeing that `firstChildOfMatchingText` is present, then we should never need to call `orElseThrow`. Also, when we `map` to some other Optional, the return type of `map` is not `@Present`.

Reply:
I agree with this Checker, but we have an issue; compilation is failing, so code must be fixed to activate Optional Checker.

Reply:
After applying this patch:
```
➜  checkstyle git:(pull-12184) ✗ git diff | cat
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java b/src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java
index fb54dba9c..cd8d99c40 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java
@@ -37,6 +37,8 @@ import java.util.stream.Collectors;
 import javax.xml.parsers.ParserConfigurationException;
 import javax.xml.transform.TransformerException;
 
+import org.checkerframework.checker.optional.qual.Present;
+
 import com.puppycrawl.tools.checkstyle.FileStatefulCheck;
 import com.puppycrawl.tools.checkstyle.api.DetailAST;
 import com.puppycrawl.tools.checkstyle.api.DetailNode;
@@ -300,12 +302,12 @@ public class JavadocMetadataScraper extends AbstractJavadocCheck {
             final DetailNode propertyNameTag = propertyNameNode.get();
             final String propertyName = getTextFromTag(propertyNameTag);
 
-            final DetailNode propertyType = getFirstChildOfMatchingText(nodeLi, TYPE_TAG)
-                .orElseThrow(() -> {
-                    return new MetadataGenerationException(String.format(
-                        Locale.ROOT, PROP_TYPE_MISSING, propertyName)
-                    );
-                });
+            final DetailNode propertyType = getFirstChildOfMatchingText(nodeLi, TYPE_TAG);
+            if (propertyType == null) {
+               throw new MetadataGenerationException(String.format(
+                                        Locale.ROOT, PROP_TYPE_MISSING, propertyName));
+            }
+
             final String propertyDesc = DESC_CLEAN.matcher(
                     constructSubTreeText(nodeLi, propertyNameTag.getIndex() + 1,
                             propertyType.getIndex() - 1))
@@ -315,21 +317,23 @@ public class JavadocMetadataScraper extends AbstractJavadocCheck {
             modulePropertyDetails.setName(propertyName);
             modulePropertyDetails.setType(getTagTextFromProperty(nodeLi, propertyType));
 
-            final Optional<DetailNode> validationTypeNodeOpt = getFirstChildOfMatchingText(nodeLi,
-                VALIDATION_TYPE_TAG);
-            if (validationTypeNodeOpt.isPresent()) {
-                final DetailNode validationTypeNode = validationTypeNodeOpt.get();
+            final DetailNode validationTypeNode = getFirstChildOfMatchingText(nodeLi, VALIDATION_TYPE_TAG);
+            if (validationTypeNode != null) {
                 modulePropertyDetails.setValidationType(getTagTextFromProperty(nodeLi,
                     validationTypeNode));
             }
 
-            final String defaultValue = getFirstChildOfMatchingText(nodeLi, DEFAULT_VALUE_TAG)
-                .map(defaultValueNode -> getPropertyDefaultText(nodeLi, defaultValueNode))
-                .orElseThrow(() -> {
-                    return new MetadataGenerationException(String.format(
-                        Locale.ROOT, PROP_DEFAULT_VALUE_MISSING, propertyName)
-                    );
-                });
+            final DetailNode nodeChild = getFirstChildOfMatchingText(nodeLi, DEFAULT_VALUE_TAG);
+            final String defaultValue;
+            if (nodeChild != null) {
+                defaultValue = getPropertyDefaultText(nodeLi, nodeChild);
+            }
+            else {
+                throw new MetadataGenerationException(
+                        String.format(Locale.ROOT, PROP_DEFAULT_VALUE_MISSING, propertyName)
+                );
+            }
+
             if (!PROPERTIES_TO_NOT_WRITE.contains(defaultValue)) {
                 modulePropertyDetails.setDefaultValue(defaultValue);
             }
@@ -512,11 +516,12 @@ public class JavadocMetadataScraper extends AbstractJavadocCheck {
      * @param pattern pattern to match against
      * @return the first child node matching the condition
      */
-    private static Optional<DetailNode> getFirstChildOfMatchingText(DetailNode node,
+    private static DetailNode getFirstChildOfMatchingText(DetailNode node,
                                                                     Pattern pattern) {
         return Arrays.stream(node.getChildren())
                 .filter(child -> pattern.matcher(child.getText()).matches())
-                .findFirst();
+                .findFirst()
+                .orElse(null);
     }
 
     /**

```

This makes Checker happy other than warnings.

Reply:
this conflicting with ideas to not use `null` (in some languages it is not a keyword at all, to not let it be used at alll) at all and deal will Optional.
Such refactoring will lead us to remove Optional. Something is wrong here....

Reply:
If the goal is to activate the checker anyhow, then `@SuppressWarnings("optional:method.invocation.invalid")` will suppress the violations and build will pass.

Reply:
>Such refactoring will lead us to remove Optional. Something is wrong here....

The issue is with the method, since we treat the return value as `@Present` in two cases, and `@MaybePresent` in one case.

Reply:
We could also just use optional inline as opposed to the method, and treat each optional separately, with the two that should throw exception as `@Present`

Reply:
TLDR;
- if we annotate with `@Present`, absence of value is exceptional
- if we annotate with `@MaybePresent` (or no annotation, as this is default), absence is not exceptional

Reply:
> absence of value is exceptional

Is the "value" in this sentence the possible value inside `Optional` or are we talking about `Optional` itself?

Reply:
I don't think annotating with `@Present` and `@MaybePresent` will work here. `java.util.Optional#map(..)` returns an optional of type `@MaybePresent` and `findFirst()` also returns the same. So until we annotate those methods to return of type `@Present` we would get the error.

Reply:
> > absence of value is exceptional
> 
> Is the "value" in this sentence the possible value inside `Optional` or are we talking about `Optional` itself?

I think "value" inside the optional.

Reply:
> I don't think annotating with `@Present` and `@MaybePresent` will work here. `java.util.Optional#map(..)` returns an optional of type `@MaybePresent` and `findFirst()` also returns the same. So until we annotate those methods to return of type `@Present` we would get the error.

The issue is that we use the return value if the method as two different “types”. We can remove method, inline code, and use the appropriate annotation.

Reply:
I am unable to get what you are trying to say, checker will flag even if we use the wrong annotation. eg-
```java
@Present final Optional<String> optional =
        anyThingThatWillReturnOptional(..)
        .map(..);
```
`map(..)` will return `@MaybePresent` and our variable is annotated with `@Present`, the checker will flag this.

Reply:
Can we suppress without annotation?

Reply:
We can skip this file by using `-AskipDefs`

Reply:
Ok, let’s not block this PR. Let’s:
1. Open an issue for further discussion
2. Add suppression by file mentioned at https://github.com/checkstyle/checkstyle/pull/12184#discussion_r967801525
3. Add comment with “until…” above suppression

Reply:
Sounds good